action/tx_container.go

```diff
@@ -11,25 +11,131 @@ import (
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
 	"github.com/iotexproject/go-pkgs/hash"
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
 	"github.com/pkg/errors"
 )
 
 var (
 	_ TxContainer = (*txContainer)(nil)
+	_ Envelope    = (*txContainer)(nil)
 )
 
 type txContainer struct {
-	raw []byte
-	tx  *types.Transaction
+	chainID uint32
+	raw     []byte
+	tx      *types.Transaction
+}
+
+func (etx *txContainer) Version() uint32 {
+	txType, _ := convertEthTxType(etx.tx.Type())
+	return uint32(txType)
+}
+
+func (etx *txContainer) ChainID() uint32 {
+	return etx.chainID
+}
+
+func (etx *txContainer) Nonce() uint64 {
+	return etx.tx.Nonce()
+}
+
+func (etx *txContainer) Gas() uint64 {
+	return etx.tx.Gas()
+}
+
+func (etx *txContainer) GasPrice() *big.Int {
+	return etx.tx.GasPrice()
+}
+
+func (etx *txContainer) EffectiveGasPrice(baseFee *big.Int) *big.Int {
+	// TODO
+	return nil
+}
+
+func (etx *txContainer) AccessList() types.AccessList {
+	return etx.tx.AccessList()
+}
+
+func (etx *txContainer) GasTipCap() *big.Int {
+	return etx.tx.GasTipCap()
+}
+
+func (etx *txContainer) GasFeeCap() *big.Int {
+	return etx.tx.GasFeeCap()
+}
+
+func (etx *txContainer) BlobGas() uint64 {
+	return etx.tx.BlobGas()
+}
+
+func (etx *txContainer) BlobGasFeeCap() *big.Int {
+	return etx.tx.BlobGasFeeCap()
+}
+
+func (etx *txContainer) BlobHashes() []common.Hash {
+	return etx.tx.BlobHashes()
+}
+
+func (etx *txContainer) BlobTxSidecar() *types.BlobTxSidecar {
+	return etx.tx.BlobTxSidecar()
+}
+
+func (etx *txContainer) Value() *big.Int {
+	return etx.tx.Value()
+}
+
+func (etx *txContainer) To() *common.Address {
+	return etx.tx.To()
+}
+
+func (etx *txContainer) Data() []byte {
+	return etx.tx.Data()
+}
+
+func (etx *txContainer) Destination() (string, bool) {
+	// TODO
+	return "", true
+}
+
+func (etx *txContainer) Size() uint32 {
+	// TODO
+	return 0
+}
+
+func (etx *txContainer) Action() Action { return etx }
```

Comment:
can we return the unfolded tx in Action()? thus will keep consistent in workingset